examples/extensions-zalando/src/main/java/org/zalando/intellij/swagger/examples/extensions/zalando/validator/zally/http/TokenService.java

```diff
@@ -0,0 +1,51 @@
+package org.zalando.intellij.swagger.examples.extensions.zalando.validator.zally.http;
+
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+
+public class TokenService {
+
+    private static final String CACHE_KEY = "token";
+
+    private static final LoadingCache<String, String> TOKEN_CACHE = CacheBuilder.newBuilder()
+            .maximumSize(1)
+            .expireAfterWrite(30, TimeUnit.MINUTES)
+            .build(
+                    new CacheLoader<String, String>() {
+                        public String load(String key) throws Exception {
+                            return createToken();
+                        }
+                    });
+
+    public static String getToken() {
+        try {
+            return TOKEN_CACHE.get(CACHE_KEY);
+        } catch (ExecutionException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static String createToken() throws IOException {
+        ProcessBuilder pb =
+                new ProcessBuilder("/bin/bash", "-c", "/usr/local/bin/ztoken");
+        pb.environment().put("LANG", "en_US.UTF-8");
+
+        Process p = pb.start();
+
+        final String token = new BufferedReader(new
+                InputStreamReader(p.getInputStream())).readLine();
+
+        if (token == null) {
```

Comment:
In case `ztoken` fails, there will be an error message wich is stored in the `token` variable. It will probably never be a `null`. Should we check the token format? (something like `[\w-.]{256-1024}`)

Reply:
The `token` is `null` in case of an error; the error is sent to `p.getErrorStream()`. So I would not add a validation for the format, if it has a value it should be a proper token.

Reply:
Path `/usr/loca/bin` was added to the `PATH` in commit https://github.com/zalando/intellij-swagger/pull/188/commits/195a1f7f465f47fcd5ba4b0c4f5152aa4ad0ad90. This will improve the odds that the plugin works out-of-the-box.